src/lib.rs

```diff
@@ -1008,20 +1008,25 @@ impl Octocrab {
     /// then obtain an installation ID, and then pass that here to
     /// obtain a new `Octocrab` with which you can make API calls
     /// with the permissions of that installation.
-    pub fn installation(&self, id: InstallationId) -> Octocrab {
+    pub fn installation(&self, id: InstallationId) -> Result<Octocrab> {
         let app_auth = if let AuthState::App(ref app_auth) = self.auth_state {
             app_auth.clone()
         } else {
-            panic!("Github App authorization is required to target an installation");
+            return Err(Error::Installation {
+                backtrace: Backtrace::generate(),
+                source: error::InstallationError::new(
+                    "Github App authorization is required to target an installation",
```

Comment:
I don't think this needs a message. I think you can just write as a display impl with snafu.

Reply:
Yeah, you're right. I could also remove the entire `InstallationError` struct since we don't need a custom message. What do you say about this?
```rust
pub enum Error {
    ...
    #[snafu(display("Installation Error: Github App authorization is required to target an installation.\n\nFound at {}", backtrace))]
    Installation { backtrace: Backtrace },
    ...
}
```

I could shorten the message if it's too long, or omit the backtrace and make the entire variant a unit:

```rust
pub enum Error {
    ...
    #[snafu(display("GitHub App authorization is required to target an installation."))]
    Installation,
    ...
}
```

Reply:
I prefer the former. The backtrace is important

Reply:
Alright. Do you prefer the first or second display message? I like the former because it keeps the style from all other error displays

Reply:
The former is good for now.